src/devtools/client/debugger/src/actions/project-text-search.js

```diff
@@ -77,16 +86,44 @@ export function searchSources(cx, query) {
     await dispatch(clearSearchResults(cx));
     await dispatch(addSearchQuery(cx, query));
     dispatch(updateSearchStatus(cx, statusType.fetching));
-    const validSources = getSourceList(getState()).filter(
-      source => !ThreadFront.isMinifiedSource(source.id) && !isThirdParty(source)
-    );
-    for (const source of validSources) {
-      if (cancelled) {
-        return;
-      }
-      await dispatch(loadSourceText({ cx, source }));
-      await dispatch(searchSource(cx, source.id, query));
-    }
+    // const validSources = getSourceList(getState()).filter(
+    //   source => !ThreadFront.isMinifiedSource(source.id) && !isThirdParty(source)
+    // );
+    // for (const source of validSources) {
+    //   if (cancelled) {
+    //     return;
+    //   }
+    //   await dispatch(loadSourceText({ cx, source }));
+    //   await dispatch(searchSource(cx, source.id, query));
+    // }
+    await ThreadFront.searchSources({ query }, matches => {
+      console.info(matches);
+      const sources = getSources(getState());
+      const bestResults = matches.filter(
+        match => !ThreadFront.isMinifiedSource(match.location.sourceId)
+      );
+      const resultsBySource = groupBy(bestResults, res => res.location.sourceId);
+      const results = Object.entries(resultsBySource).map(([sourceId, matches]) => {
```

Comment:
to be idiomatic, i'd probably pull this logic into a helper that can use the right selectors etc, but otherwise simple

```js
const results = Object.entries(resultsBySource).map(([sourceId, matches]) => mapSearchResultMatches(state, sourceId, matches)
```
```

Reply:
9a82c68